src/scent_map.h

```diff
@@ -0,0 +1,63 @@
+#ifndef SCENT_H
+#define SCENT_H
+
+#include "enums.h"
+#include "game_constants.h"
+#include "cursesdef.h"
+
+#include <array>
+
+class map;
+
+class scent_map
+{
+    protected:
+        template<typename T>
+        using scent_array = std::array<std::array<T, SEEY *MAPSIZE>, SEEX *MAPSIZE>;
+
+        scent_array<int> grscent;
+        int null_scent = 0;
```

Comment:
Yes. that's where it's used. The function returns a reference and therefor it needs some object to return even if the input coordinates are out of bounds. The concept is used in several other classes as well.